source/state_representation/tests/testJacobian.cpp

```diff
@@ -1,145 +1,117 @@
 #include "state_representation/Robot/Jacobian.hpp"
-#include <gtest/gtest.h>
 #include <fstream>
+#include <gtest/gtest.h>
 #include <unistd.h>
 
-TEST(TestCreate, PositiveNos)
-{
-	StateRepresentation::Jacobian jac("robot", Eigen::MatrixXd::Random(6,7));
-
-	EXPECT_TRUE(jac.get_nb_rows() == 6);
-	EXPECT_TRUE(jac.get_nb_cols() == 7);
-
-	bool except_thrown = false;
-	try
-	{
-		jac.set_data(Eigen::MatrixXd::Random(7,6));
-	}
-	catch(const IncompatibleSizeException& e)
-	{
-		except_thrown = true;
-	}
-	EXPECT_TRUE(except_thrown);
+TEST(TestCreate, PositiveNos) {
+  StateRepresentation::Jacobian jac("robot", Eigen::MatrixXd::Random(6, 7));
+
+  EXPECT_TRUE(jac.get_nb_rows() == 6);
+  EXPECT_TRUE(jac.get_nb_cols() == 7);
+
+  bool except_thrown = false;
+  try {
+    jac.set_data(Eigen::MatrixXd::Random(7, 6));
+  } catch (const IncompatibleSizeException& e) {
+    except_thrown = true;
+  }
+  EXPECT_TRUE(except_thrown);
```

Comment:
You could change this to   EXPECT_THROW(jac.set_data(Eigen::MatrixXd::Random(7, 6)),  Exceptions::IncompatibleSizeException);

Reply:
yes good catch I saw you doing that in your test model. Will change it